### DISCUSSION TITLE
Standalone GLSP web editor example

### DISCUSSION THREAD

**Discussion by umar-shabbir** [eclipse-glsp/glsp#1445] [General]
The [Platform Integrations](https://eclipse.dev/glsp/documentation/integrations/#platform-integrations) section in the documentation mentions links to GLSP integration examples, but the last point in the list is plain text whereas all others are links to the respective examples.

- **Creating a standalone GLSP web editor based on plain javascript**

Does it mean there is no example available for GLSP standalone integration in some javascript/typescript web app?

Though I have already looked at [workflow-standalone](https://github.com/eclipse-glsp/glsp-client/tree/master/examples/workflow-standalone) and a discussion here: https://github.com/eclipse-glsp/glsp/discussions/840

Are there any other resources that may help integrating GLSP client in an existing web app (frontend javascript/typescript), and integrating the GLSP server in an existing backend (node javascript/typescript) app ?

**Comment by planger** [comment]
Thank you for bringing that up!

The links are pointing to the integration repositories for Theia, VS Code and Eclipse. I guess we didn't put a link for the Standalone because there is no dedicated integration necessary for standalone and it is part of the [glsp-client](https://github.com/eclipse-glsp/glsp-client) repo.

But you are right, we should still link it for consistency and clarify. I'll open a PR.

And regarding examples: Yes there is an example:
https://github.com/eclipse-glsp/glsp-client/tree/master/examples/workflow-standalone

Thanks again!

**Reply by umar-shabbir** [reply]
Yes, I did see this workflow-standalone example as I have mentioned in my post. But this example part of the repo is not a minimal one and it uses a workflow-server implementation which is also not very basic.

I was looking for a minimal example, which one can start from installing the packages from npm (client, server, anything else), and the required code for lets say adding a single rectangle node on a web page. Ideally, the standalone integration in some web project should not have a dependency on the eclipse-glsp-examples repository, rather it should drive the implementation from the client and server packages.

**Reply by planger** [reply]
I see, I don't think we have a dedicated minimal standalone example, beyond the workflow example.
The project templates are all geared towards a specific integration (Theia, VS Code, Eclipse).

Having said that, however, the VS Code integration essentially contains a minimal example to some degree:
https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/node-json-vscode/tasklist-vscode/webview

Also, there is the this repo: https://github.com/Sakrafux/glsp-browser-standalone-integration/tree/main
Contains browser-only examples that may be helpful.

Beyond that, I'm not aware of existing examples.
